Project/Source/Components/UI/ComponentToggle.h

```diff
@@ -1,22 +1,41 @@
 #pragma once
 
 #include "Components/Component.h"
+#include "UI/Interfaces/IMouseClickHandler.h"
+#include "Math/float4.h"
 
-class ComponentToggle : public Component {
+class ComponentImage;
+
+class ComponentToggle : public Component
+	, public IMouseClickHandler {
 public:
 	REGISTER_COMPONENT(ComponentToggle, ComponentType::TOGGLE, false);
 
 	~ComponentToggle();
 
 	void Init() override;
 	void DuplicateComponent(GameObject& obj) override;
+	void OnClicked() override; // Action to perform when the button is clicked
 	void Save(JsonValue jComponent) const override;
 	void Load(JsonValue jComponent) override;
+	void OnEditorUpdate() override;
 
 	void OnValueChanged();
-	bool IsOn();
+	bool IsOn() const;
 	void SetOn(bool b);
+	ComponentImage* GetEnabledImage() const;
+	void SetEnabledImageObj(UID enabledImageObjID_);
+
+	bool IsClicked() const;
+	void SetClicked(bool clicked_);
+
+	float4 GetTintColor() const;  // Returns the correspondant color of the current state
+	float4 GetClickColor() const; // Returns colorClicked
 
 private:
-	bool isOn = false;
+	bool isOn = true;
```

Comment:
I suggest renaming the "isOn" variable to "checked".

Reply:
isChecked maybe, but no big deal

Reply:
I agree